vault/seal.go

```diff
@@ -329,14 +329,9 @@ type SealConfig struct {
 	VerificationProgress [][]byte `json:"-"`
 }
 
-// Validate is used to sanity check the seal configuration
-func (s *SealConfig) Validate() error {
-	if s.SecretShares < 1 {
-		return errors.New("shares must be at least one")
-	}
-	if s.SecretThreshold < 1 {
-		return errors.New("threshold must be at least one")
-	}
+// baseValidate is used as a shared base between `Validate` and `ValidateRecovery`
+// functions
+func (s *SealConfig) baseValidate() error {
 	if s.SecretShares > 1 && s.SecretThreshold == 1 {
```

Comment:
```suggestion
	if s.SecretShares > 1 && s.SecretThreshold < 2 {
```

(In particular, with the removal of `s.SecretThreshold < 1` check above, a recovery key could've come in and set shares > 1 but threshold == 0, which would be bad)